src/plugins/vis_types/vega/public/vega_view/vega_base_view.js

```diff
@@ -293,12 +293,8 @@ export class VegaBaseView {
   }
 
   updateVegaSize(view, dimensions) {
```

Comment:
I believe I found the way to fix also the scrolls:
the above method `async resize` is called by the EUIResizeObserver with the actual dimensions from the container.
if you add this dimension to the resize function and pass it to the `.updateVegaSize(this._view, dimension);` then you can use them correctly.

In fact, if you check there is no use of `updateVegaSize` with 2 arguments, and the `dimensions` argument is always undefined. I believe is just missed here

Reply:
Yeah good catch! I fixed this in 81a7703df12ac7ed2fba73436b4bd380fff2399e, which actually fixed #194861.

Reply:
One other issue I found is the initial rendering. In rare cases, particularly smaller panels, `~300px` height, the initial render creates a scrollbar. I found this is due to the `dimensions` being `undefined` but the `clientHeight` is automatically rounded up by `0.5px` causing a `0.5px` overflow.

I think a simple fix would just be to always take off `1px`. WDYT?

Reply:
Firefox works perfectly, but chrome/safari still shows some issues. I believe this is fine for now. I checked the panels in dashboard and there is no change here so good to go

Reply:
Yeah I see this bad dimensions in sahari, might be due to the eui `ResizeObserver`.